lib/prism/parse_result/newlines.rb

```diff
@@ -54,101 +54,101 @@ def visit_if_node(node)
       # Permit statements lists to mark newlines within themselves.
       def visit_statements_node(node)
         node.body.each do |child|
-          child.newline!(@lines)
+          child.newline_flag!(@lines)
         end
         super(node)
       end
     end
   end
 
   class Node
-    def newline? # :nodoc:
-      @newline ? true : false
+    def newline_flag? # :nodoc:
```

Comment:
It's not ideal to rename this one and have the common flag be called `newline?`, because the latter is the "potential but not necessary newline" while the former is the real newline flag, i.e. "this node is the sole node on that line with the flag".
How about renaming the common flag as POTENTIAL_NEWLINE or so?